semgrep-core/tests/go/metavar_typed_struct.go

```diff
@@ -0,0 +1,24 @@
+func foo() {
+	var a http.Server
+	//ERROR: match
+	a.ListenAndServe()
+
+	b := http.Server{}
+	//ERROR: match
+	b.ListenAndServe()
+
+	var c = http.Server{}
+	//ERROR: match
+	c.ListenAndServe()
+
+	d := http.Server{Addr:":8080"}
+	//ERROR: match
+	d.ListenAndServe()
+
+	var e = http.Server{Addr:":8080",ReadTimeout:10}
+	//ERROR: match
+	e.ListenAndServe()
+
+	var f = http.Client{}
+	f.ListenAndServe()
```

Comment:
Although it really doesn't do anything AFAICT, we tend to annotate these lines with `//OK:` to stress that there should be no match there. Not a big deal.